app/lib/actions/pulp/repository/sync.rb

```diff
@@ -24,6 +25,10 @@ def invoke_external_task
               # set threads per sync
               sync_options[:num_threads] ||= SETTINGS[:katello][:pulp][:sync_threads]
             end
+            if input[:source_url]
+              sync_options[:feed] = input[:source_url]
+            end
```

Comment:
nitpick, you might condense this to one line   `sync_options[:feed] = input[:source_url] if input[:source_url]`